### PR TITLE
Avoid duplicate plugin assertion.

### PR DESCRIPTION
**Pull request by rwjblue**
This is a fairly rare issue, but when it _does_ happen it is quite difficult to debug.

The specific setup is:

* addon A depends on ember-compatibility-helpers
* addon A has the following in its `index.js`:

```js
module.exports = {
  name: require('./package').name,

  options: {
    m3: true,
    babel: {
      loose: true,
    },
  },
};
```

* addon B depends on addon A (**not** a dev dep)

In this scenario, addon A will be instantiated twice. Once for addon B's dummy app and again for addon B's direct dependency. Since addon A had a _mutable object_ on its prototype (yes, you know whats coming now) ember-compatibility-helpers adds a `plugins` array to it and pushes into it. Even though it sets a local flag saying that `_registeredWithBabel`, that flag is _per instance_ of ember-compatibility-helpers and in our scenario we actually have **two** instances (therefore making our check moot).

The fixes here, prevent adding our comparison and debug plugins twice by checking if they are already present before pushing (it also adds a name property, which helps as well).

### CODE REVIEW

**Comment by rwjblue** [comment]
FYI - I wanted to use ember-cli-babel-plugin-helpers to do this, but it does not work when you want to search for a specific name within the plugins array. I created https://github.com/dfreeman/ember-cli-babel-plugin-helpers/issues/5 to track that.

**Comment by rwjblue** [comment]
CI is getting trolled by:

* running in Node 6
* no lockfile
* someone has updated to `async` functions without a major version bump

**Comment by makepanic** [comment]
This fixes the issue for us when upgrading ember-data to 3.14 in a monorepo.

**Comment by pzuraq** [comment]
Let's bump to Node 8 and release a new major

**Comment by rwjblue** [comment]
👍

**Comment by omairvaiyani** [comment]
Fixed our issue, the culprit addon was `@ember/ordered-set`

**Comment by gabrielcsapo** [comment]
Is this something we can get merged? We are getting duplicate addons in our product.

File: index.js [hunk 1]
```diff
@@ -47,12 +47,22 @@ module.exports = {
...
     parentOptions.babel = parentOptions.babel || {};
-
     const plugins = parentOptions.babel.plugins = parentOptions.babel.plugins || [];
+
     const comparisonPlugin = this._getComparisonPlugin(this.emberVersion);
     const debugPlugin = this._getDebugPlugin(this.emberVersion, this.parentChecker);
 
-    plugins.push(comparisonPlugin, debugPlugin);
+    if (this._usingBabel7) {
+      if (!plugins.find(p => Array.isArray(p) && p[2] === comparisonPlugin[2])) {
```
Comment (gabrielcsapo) [inline index.js:56]: errors might be the reason you are getting trolled in CI.

**Comment by gabrielcsapo** [comment]
Looks like the async is coming from broccoli-rollup which is from ember-data

```
  - stack: /home/travis/build/pzuraq/ember-compatibility-helpers/node_modules/broccoli-rollup/dist/index.js:31
603    async build() {
```

```
=> Found "broccoli-rollup@2.1.1"
info Reasons this module exists
   - "ember-data" depends on it
   - Hoisted from "ember-data#broccoli-rollup"
```

**Comment by gabrielcsapo** [comment]
This might need to be rebased to get the CI fixes and rerun to get passing.

**Comment by rwjblue** [comment]
Rebased and pushed...

**Comment by rwjblue** [comment]
OK, finally green (thanks all). 

@pzuraq - Mind merge + releasing (I'm on mobile)?

**Comment by allthesignals** [comment]
I'm running into this after upgrading ember. I've already installed the latest ember-compatibility-helpers and ordered-set. Here's the log: 

```
Build Error (broccoli-persistent-filter:Babel > [Babel: @ember/ordered-set]) in @ember/ordered-set/index.js

Duplicate plugin/preset detected.
If you'd like to use two separate instances of a plugin,
they need separate names, e.g.

  plugins: [
    ['some-plugin', {}],
    ['some-plugin', {}, 'some unique name'],
  ]

Duplicates detected are:
[
  {
    "alias": "/Users/wmattgardner/labs-zola/node_modules/@ember/ordered-set/node_modules/ember-compatibility-helpers/comparision-plugin.js",
    "options": {
      "emberVersion": "3.24.1",
      "root": "/Users/wmattgardner/labs-zola",
      "name": "@ember/ordered-set"
    },
    "dirname": "/Users/wmattgardner/labs-zola",
    "ownPass": false,
    "file": {
      "request": "/Users/wmattgardner/labs-zola/node_modules/@ember/ordered-set/node_modules/ember-compatibility-helpers/comparision-plugin.js",
      "resolved": "/Users/wmattgardner/labs-zola/node_modules/@ember/ordered-set/node_modules/ember-compatibility-helpers/comparision-plugin.js"
    }
  },
  {
    "alias": "/Users/wmattgardner/labs-zola/node_modules/@ember/ordered-set/node_modules/ember-compatibility-helpers/comparision-plugin.js",
    "options": {
      "emberVersion": "3.24.1",
      "root": "/Users/wmattgardner/labs-zola",
      "name": "@ember/ordered-set"
    },
    "dirname": "/Users/wmattgardner/labs-zola",
    "ownPass": false,
    "file": {
      "request": "/Users/wmattgardner/labs-zola/node_modules/@ember/ordered-set/node_modules/ember-compatibility-helpers/comparision-plugin.js",
      "resolved": "/Users/wmattgardner/labs-zola/node_modules/@ember/ordered-set/node_modules/ember-compatibility-helpers/comparision-plugin.js"
    }
  },
  {
    "alias": "/Users/wmattgardner/labs-zola/node_modules/@ember/ordered-set/node_modules/ember-compatibility-helpers/comparision-plugin.js",
    "options": {
      "emberVersion": "3.24.1",
      "root": "/Users/wmattgardner/labs-zola",
      "name": "@ember/ordered-set"
    },
    "dirname": "/Users/wmattgardner/labs-zola",
    "ownPass": false,
    "file": {
      "request": "/Users/wmattgardner/labs-zola/node_modules/@ember/ordered-set/node_modules/ember-compatibility-helpers/comparision-plugin.js",
      "resolved": "/Users/wmattgardner/labs-zola/node_modules/@ember/ordered-set/node_modules/ember-compatibility-helpers/comparision-plugin.js"
    }
  }
]
```

**Pull request merged**